architecture/README.md

```diff
@@ -1 +1,147 @@
-# Architecture
\ No newline at end of file
+# Architecture
+
+The Open Sovereign AI Cloud solution gives cloud providers a complete platform
+to offer self-service provisioning of compute infrastructure, including VMs and
+OpenShift clusters, integrated with the cloud provider’s existing infrastructure
+components. It features a fulfillment workflow that is powerful enough to handle
+complex cluster deployment while also being flexible enough to accommodate
+future expansion into other types of deployments.
+
+## Templates
+
+Self-service provisioning in OSAC is built around a concept of Templates.
+Whether an end user wants to provision a VM, a cluster, or something else,
+they'll be presented with a selection of templates from which to choose. Upon
+selecting a template, they'll provide the required input, and then the system
+will proceed with a provisioning workflow.
+
+The CSP can define their own templates, including:
+
+* what input each template requires
+* what infrastructure it provisions
+* how it provisions that infrastructure
+
+Ansible makes that possible, because Ansible
+[Roles](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html)
+can be used as Templates. Metadata such as [argument
+validation](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html#role-argument-validation)
+describes how to use a template, while the contents of the Role itself codify
+the implementation of provisioning.
```

Comment:
The phrasing "...because Ansible Roles can be used as Templates..." seems a bit weak: it suggests that maybe *other* things can be used Templates, too. I think we should make this more explicit: "A Template is an Ansible Role ...".